### DISCUSSION TITLE
mqtt.publish not working from bubble-card

### DISCUSSION THREAD

**Discussion by Fireblade900rr** [Clooos/Bubble-Card#734] [Questions about config, custom styles and templates]
I tried to publsih a mqtt value with tap-action (which is possible with std button card) but nothing happens.

Heres a part of my code, can anyone tell me whats wrong ?

Its not working to paste the code properly, its no identation error, it just happens nothing

`type: custom:bubble-card
card_type: button
name: Heißwassermodus
icon: mdi:water-opacity
tap_action:
  service: mqtt
  action: none
  target: {}
button_action:
  tap_action:
    action: none
    service: mqtt.publish
  double_tap_action:
    action: none
  hold_action:
    action: none
entity: sensor.nibe_heisswassermodus
sub_button:
  - entity: sensor.nibe_heisswassermodus
    name: Sparen
    show_name: true
    show_state: false
    icon: mdi:spa
    tap_action:
      action: call-service
      service: mqtt.publish
      service_data:
        topic: homeassistant/nibe/modbus/47041
        payload: '0'
      target: {}`

**Comment by Clooos** [comment]
Hi, you can try with `data:` instead of `service_data:`, this should work.

**Reply by Fireblade900rr** [reply]
Yes ! Thats it. I tried before with data:  but with no success, maybe something else has been wrong. Thanks !
